cmd/kubeadm/app/util/version_test.go

```diff
@@ -113,17 +112,15 @@ func TestVersionFromNetwork(t *testing.T) {
 		ErrorExpected        bool
 	}
 
-	currentVersion := normalizedBuildVersion(constants.CurrentKubernetesVersion.String())
-
 	cases := map[string]T{
 		"stable":     {"stable-1", "v1.4.6", false, false}, // recursive pointer to stable-1
 		"stable-1":   {"v1.4.6", "v1.4.6", false, false},
 		"stable-1.3": {"v1.3.10", "v1.3.10", false, false},
 		"latest":     {"v1.6.0-alpha.0", "v1.6.0-alpha.0", false, false},
 		"latest-1.3": {"v1.3.11-beta.0", "v1.3.11-beta.0", false, false},
-		"empty":      {"", currentVersion, true, false},
-		"garbage":    {"<?xml version='1.0'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message></Error>", currentVersion, true, false},
-		"unknown":    {"The requested URL was not found on this server.", currentVersion, true, false},
+		"empty":      {"", "", true, true},
+		"garbage":    {"<?xml version='1.0'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message></Error>", "", true, true},
+		"unknown":    {"The requested URL was not found on this server.", "", true, true},
```

Comment:
i don't think these changes are valid.
`kubernetesReleaseVersion()` does fallback to the version in a constant:
https://github.com/kubernetes/kubernetes/blob/67d928acdc351d3d4f9e20a92cee4452a26ed0c4/cmd/kubeadm/app/util/version.go#L109

Reply:
That case applies when we have been able to parse the version with `kubeReleaseLabelRegex.MatchString(versionLabel)` section but not able to find that version. With the change in the regex, we would only be parsing strings that start with 'k8s-master`, 'latest` and `stable`. For everything else (i.e. empty, garbage etc), the control will go to: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/version.go#L125 resulting in error. Please let me know if that assumption is not correct.

Reply:
what seems wrong here is that both `FetcherErrorExpected` and `ErrorExpected` now always match in the test cases. if the regex now catches the badly formatted labels, then perhaps it makes sense to only have a single `ErrorExpected`.

but we still need a test that validates the fallback to the constant (local version) in case the network is not available. that was the purpose of the `currentVersion` variable used as a test case result.

Reply:
Yeah, sure, makes sense. I would leave these tests as is and will add another test, that will check the fall back flow and will return back to currenVersion. Does that sounds good?

Reply:
ok